frontend/src/main/components/Commons/PagedProfitsTable.js

```diff
@@ -0,0 +1,107 @@
+                {
+                    Header: "Health",
+                    accessor: (row) => `${row.avgCowHealth.toFixed(2) + '%'}`
+                },
+                {
+                    Header: "Cows",
+                    accessor: "numCows",
+                },
+            ],
+        []);
+    const memoizedDates = React.useMemo(() => profitsForTable, [profitsForTable]);
+    // Stryker restore ArrayDeclaration
+
+    return (
+        <div>
+            <Card>
+                <Card.Header as="h5">
+                    Profits
+                </Card.Header>
+                <Card.Body>
+                    <Card.Title>
+                        You will earn profits from milking your cows everyday at 4am.
+                    </Card.Title>
+                    <OurTable
+                        data={memoizedDates}
+                        columns={memoizedColumns}
+                        testid={"ProfitsTable"}
+                    />
+                </Card.Body>
+
+
+                {/* Pagination component */}
+                <div data-testid="PageNavBottom" className="d-flex justify-content-center">
+                    <Pagination>
+                        <Pagination.First onClick={() => onPageChange(0)} data-testid="PageNavBottom-btn-first"/>
+                        <Pagination.Prev onClick={() => onPageChange(pageable.pageNumber - 1)}
+                                         data-testid="PageNavBottom-btn-prev"/>
+
+                        {/* Loop through the nearby pages */}
+                        {Array.from({length: totalPages}, (_, index) => {
+                            if (
+                                index >= pageable.pageNumber - 1 &&
+                                index <= pageable.pageNumber + 1
+                            ) {
+                                return (
+                                    <Pagination.Item
+                                        key={index}
+                                        // Stryker disable next-line all : hard to check parent attribute unless accessing DOM
+                                        active={index === pageable.pageNumber}
+                                        onClick={() => onPageChange(index)}
+                                        data-testid={`PageNavBottom-btn-item-${index}`}
+                                    >
+                                        {index + 1}
+                                    </Pagination.Item>
+                                );
+                            }
+                            return null;
+                        })}
+
+                        <Pagination.Next onClick={() => onPageChange(pageable.pageNumber + 1)}
+                                         data-testid="PageNavBottom-btn-next"/>
+                        <Pagination.Last onClick={() => onPageChange(totalPages - 1)}
+                                         data-testid="PageNavBottom-btn-last"/>
+                    </Pagination>
```

Comment:
I won't hold up the PR for this; instead, I'll make a refactoring task for the next class.  But it would be nice to factor this out into a Pagination component that would be compact and reusable across multiple parts of the application (jobs, profits, etc.)